src/sentry/static/sentry/app/components/group/sentryAppExternalIssueForm.tsx

```diff
@@ -123,12 +176,84 @@ export class SentryAppExternalIssueForm extends React.Component<Props> {
     }
   }
 
+  handleFieldChange = async (id: string, _value: FieldValue) => {
+    const config = this.state;
+
+    let requiredFields = config.required_fields || [];
+    let optionalFields = config.optional_fields || [];
+
+    const fieldList: FieldFromSchema[] = requiredFields.concat(optionalFields);
+
+    //could have multiple impacted fields
+    const impactedFields = fieldList.filter(({depends_on}) => {
+      if (!depends_on) {
+        return false;
+      }
+      // must be dependent on the field we just set
+      return depends_on.includes(id);
+    });
+
+    //reset all impacted fields first
+    impactedFields.forEach(impactedField =>
+      this.model.fields.delete(impactedField.name || '')
+    );
+
+    //iterate through all the impacted fields and get new values
+    for (const impactedField of impactedFields) {
+      const choices = await this.makeExternalRequest(impactedField, '');
```

Comment:
If you make multiple `impactedFields`, this is going to perform the requests serially because of the await.

You might want something like:

```
Promise.all(impactedFields.map(field => this.makeExternalRequest(field, ''))).forEach(choices => {
    ...
});
```

Should your setState be inside of this? Right now it's only called once

Reply:
@billyvg Yea, I can make that change. I put set state at the end so that fields the options would be populated at the same time but I don't think it matters that much. I can put setState inside of there as long as I make sure to use a fresh state.